### DISCUSSION TITLE
Looking for a way to dynamically display the <title> in the index.html bases on the routes/pages

### DISCUSSION THREAD

**Discussion by markteekman** [hannoeru/vite-plugin-pages#162] [Q&A] [answered]
Hi all,

I'm so stocked that I found this plugin, it will make the lives of our team members that much easier! It's so cool to see that you don't need Nuxt for this kind of stuff 😄I don't know if this is a stupid question or at all related to this plugin. But I'm trying to dynamically set the page titles based on the pages. Is there a way to do this? I've tried adding this approach in my routers index.js, which of course doesn't render a valid title, but it does at least display the name:

```js
router.beforeEach((to, from, next) => {
  document.title = `${to.name}`
  next('/')
})
```

However, now no content is loaded and I get blank pages. I would like to have something  like this in the index.html:

```html
<title>Project name - route.meta.title</title>
```

Hope someone can help and thanks in advance! 😊

**Comment by jbaubree** [comment, marked as answer]
Hi, i think this is what you want to get :

```
const router = createRouter({
  history: createWebHistory(),
  routes,
})

const APP_NAME = 'Project name'
router.beforeEach((to, _from, next) => {
  document.title = `${APP_NAME} - ${to.name?.toString()}`
  next()
})
```

**Reply by markteekman** [reply]
@jbaubree you are a lifesaver, thank you so much for the solution! I guess it's more a of a router thing, but I'm glad this works! 😃

**Comment by markteekman** [comment]
For anyone interested, I went with the following setup based on @jbaubree's suggestion:

```js
function capitalizeFirstLetter(string) {
  return string.charAt(0).toUpperCase() + string.slice(1);
}

const APP_NAME = 'Project name'
router.beforeEach((to, _from, next) => {
  document.title = `${APP_NAME} - ${capitalizeFirstLetter(to.name?.toString().replace(/-/g, ' '))}`
  next()
})
```

Which will result to 'Project name - My home page' for example on a page called `my-home-page.vue` .

**Comment by hannoeru** [comment]
https://vueuse.org/core/usetitle/#usetitle

**Reply by markteekman** [reply]
Thanks for the addition @hannoeru, that's just what I needed! This is a nice way to set a custom homepage title, which would else be 'Index' 🤔
